src/alert.cpp

```diff
@@ -2414,4 +2414,22 @@ namespace {
 			, m_v6_num_nodes, m_v6_nodes_idx);
 	}
 
+	block_uploaded_alert::block_uploaded_alert(aux::stack_allocator& alloc, torrent_handle h
+		, tcp::endpoint const& ep, peer_id const& peer_id, int block_num
+		, int piece_num)
+		: peer_alert(alloc, h, ep, peer_id)
+		, block_index(block_num)
+		, piece_index(piece_num)
+	{
+		TORRENT_ASSERT(block_index >= 0 && piece_index >= 0);
```

Comment:
the second comparison here need to be against ``piece_index_t{0}`` instead of just 0.
If you build with ``TORRENT_NO_DEPRECATE`` defined, this should be an error for you.